### PR TITLE
Fix abandoned children nondeterminism if they complete

### PR DESCRIPTION
**Pull request by Sushisource**
* Also fix arch diagram & compiler err changes / rust ver in CI

## Why?
Abandoned children can complete/fail/timeout/terminate & that's fine. Don't barf.

## Checklist

1. Closes https://github.com/temporalio/sdk-core/issues/568

2. How was this tested:
IT & UT

3. Any docs updates needed?
nawp

### CODE REVIEW

File: core/src/worker/workflow/machines/child_workflow_state_machine.rs [hunk 1]
```diff
@@ -79,6 +82,15 @@ fsm! {
     Cancelled --(Cancel) --> Cancelled;
     Cancelled --(ChildWorkflowExecutionCancelled,
         on_child_workflow_execution_cancelled) --> Cancelled;
+    // Completions of any kind after cancellation are acceptable for abandoned children
+    Cancelled --(ChildWorkflowExecutionCompleted(Option<Payloads>),
```
Comment (cretz) [inline core/src/worker/workflow/machines/child_workflow_state_machine.rs:86]: Wonder if one day you may want to expand this macro format to build in "guards" so you can just add an if-abandoned conditional

File: core/src/worker/workflow/machines/child_workflow_state_machine.rs [hunk 2]
```diff
@@ -120,6 +132,14 @@ pub(super) struct Cancelled {
     seen_cancelled_event: bool,
 }
 
+fn completion_of_not_abandoned_err() -> WFMachinesError {
+    WFMachinesError::Nondeterminism(
+        "Child workflows which don't have the ABANDON cancellation type cannot complete after \
```
Comment (cretz) [inline core/src/worker/workflow/machines/child_workflow_state_machine.rs:137]: Hrmm, I figured `ABANDON` was a parent-close-only thing. Arguably this is a lang bug if they send you a cancel-after-complete of any child/activity regardless of abandon setting. What situation/lang is sending you cancel of completed-yet-abandoned children?

File: core/src/worker/workflow/machines/child_workflow_state_machine.rs (hunk 1 above)
Comment (Sushisource) [reply core/src/worker/workflow/machines/child_workflow_state_machine.rs:86]: Maybe. I don't want to make the DSL to complex when you can handle that in the handler easily.

File: core/src/worker/workflow/machines/child_workflow_state_machine.rs (hunk 2 above)
Comment (Sushisource) [reply core/src/worker/workflow/machines/child_workflow_state_machine.rs:137]: Lang shouldn't have to think about that, that's the point of core. And this isn't cancel after-complete, this is complete after cancel, which is outside of SDK control entirely. The activity never sees a cancel, and thus never knows it shouldn't complete.

**Review by cretz — APPROVED** [review]

**Pull request merged**